test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -85,13 +90,33 @@ session_simulator::rollback_transaction()
     _txn_running = false;
 }
 
-void
-session_simulator::commit_transaction()
+int
+session_simulator::commit_transaction(const std::string &config)
 {
     /* Make sure that the transaction from this session is running. */
     assert(_txn_running);
 
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    ts_manager->parse_config(config, config_map);
```

Comment:
What happens if the config string is empty?

Reply:
If the string is empty the config_map will also be empty and the rest of the conditions won't apply. There are some cases where the generated call log has the string (null) instead of being empty. I'll add a case for handling this in parse_config.